onnxruntime/core/optimizer/qdq_transformer/selectors_actions/shared/utils.h

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <string>
+#include "core/graph/basic_types.h"
+#include "core/providers/nnapi/nnapi_builtin/nnapi_lib/NeuralNetworksTypes.h"
+#include "../qdq_selectors.h"
+#include "../../../selectors_actions/helpers.h"
+
+namespace onnxruntime {
+
+class GraphViewer;
+class Node;
+
+namespace QDQ {
+
+struct Selector {
+  using OpVersionsMap = std::unordered_map<std::string, std::vector<ONNX_NAMESPACE::OperatorSetVersion>>;
+
+  Selector(const OpVersionsMap& ops_and_versions_in,
+           std::unique_ptr<BaseSelector> selector_in)
+      : op_versions_map{ops_and_versions_in},
+        selector{std::move(selector_in)} {}
+
+  OpVersionsMap op_versions_map;
+  std::unique_ptr<BaseSelector> selector;
+
+  ORT_DISALLOW_COPY_AND_ASSIGNMENT(Selector);
+};
+
+class Selectors {
+ public:
+  Selectors() = default;
+
+  Selectors(Selectors&& rhs) noexcept
+      : selectors_set_{std::move(rhs.selectors_set_)} {}
+
+  void RegisterSelector(const Selector::OpVersionsMap& ops_and_versions_in,
+                        std::unique_ptr<BaseSelector> selector_in);
+
+  const std::unordered_set<std::unique_ptr<Selector>>& SelectorsSet() const {
+    return selectors_set_;
+  }
+
+  ORT_DISALLOW_COPY_AND_ASSIGNMENT(Selectors);
+
+ private:
+  std::unordered_set<std::unique_ptr<Selector>> selectors_set_;
+};
+
+class SelectorManager {
+ public:
+  SelectorManager() = default;
+
+  virtual ~SelectorManager() = default;
+
+  void InitializeSelectorsMap();
+
+  void CreateSelectors();
+
+  const std::unordered_map<const Node*, std::unique_ptr<BaseSelector>> GetQDQSelectors(const GraphViewer& graph_viewer) const;
```

Comment:
Why do we need to return a map<Node*, BaseSelector>? Do we need to use the qdq selector later? Or we only need the result from QDQ selector?
Can't we have a function just return a std::vector<NodeGroup>? Such as
```
const std::vector<NodeGroup> GetQDQSelections(const GraphViewer& graph_viewer) const;
```

Reply:
Yes.. returning a map of target node* to corresponding selectors might be a bit weird... 
My original idea was to return this map with selectors needed and corresponding nodes, then for each (node*, selector) pair, call selector->GetQDQSelection(graph_viewer, *node) to get group of qdq node selections where needed.

I can change to directly return a vector of qdq node selections here.